carmajava/signal_plugin/src/test/java/gov/dot/fhwa/saxton/carma/signal_plugin/MockCollisionChecker.java

```diff
@@ -0,0 +1,51 @@
+package gov.dot.fhwa.saxton.carma.signal_plugin;
+
+import java.util.List;
+
+import gov.dot.fhwa.saxton.carma.signal_plugin.ead.INodeCollisionChecker;
+import gov.dot.fhwa.saxton.carma.signal_plugin.ead.trajectorytree.Node;
+
+public class MockCollisionChecker implements INodeCollisionChecker {
+
+    Node ncvStartNode, ncvEndNode;
+    
+    @Override
+    public boolean hasCollision(List<Node> trajectory) {
+        Node p1 = trajectory.get(0);
+        Node q1 = trajectory.get(1);
+        Node p2 = ncvStartNode;
+        Node q2 = ncvEndNode;
+        if(p1 == null || q1 == null || p2 == null || q2 == null) return false;
+        int o1 = orientation(p1, q1, p2); 
+        int o2 = orientation(p1, q1, q2); 
+        int o3 = orientation(p2, q2, p1); 
+        int o4 = orientation(p2, q2, q1);
+        if (o1 != o2 && o3 != o4) return true;  
+        if (o1 == 0 && onSegment(p1, p2, q1)) return true; 
+        if (o2 == 0 && onSegment(p1, q2, q1)) return true; 
+        if (o3 == 0 && onSegment(p2, p1, q2)) return true; 
+        if (o4 == 0 && onSegment(p2, q1, q2)) return true;
+        return false;
+    }
+    
+    public void setPredictedTrajectory(Node startNode, Node endNode) {
+        ncvStartNode = startNode;
+        ncvEndNode = endNode;
+    }
+    
+    
+    private boolean onSegment(Node p, Node q, Node r) {
+        return q.getTime()     >= Math.max(p.getTime(), r.getTime())     &&
```

Comment:
I think you have the signs of this check flipped
https://www.geeksforgeeks.org/check-if-two-given-line-segments-intersect/

Reply:
You are right...